### ISSUE TITLE
Idea: Optional harder difficulty settings and immersive items

### ISSUE THREAD

**Issue by sefldron** [SilentChaos512/SilentGems#199]
Minecraft Version: 1.12
Silent's Gems Version: 2.4.5
Silent Lib Version: 2.2.7
Silent's Gems: Extra Parts Version: 1.3.2

   This is an idea I believe would be great for you. Add a feature that would allow the tools and armor from all parts to break permanently if you want them to. On top of this allow an option where the repairs will cost X amount of levels per material used or ex point per 100-200 durability repaired? .If the item is fully repaired then it would half or quarter the Ex cost for decoration. Also I think it would be cool to see an option where repairs need to be done on a chaos anvil which could only be built by smelting soft chaos essence in a furnace to form it into an type of ingot and then building the anvil either like a vanilla anvil with formed chaos blocks and ingots or have your own repair table similar to a crafting table. It could break over time or stay forever. I love the mod the way it is as well but I feel like it is too easy to keep an item alive. Kinda takes away from that "oh crap my protection 3 titanium everything might break soon" feeling when you're far from home. The optional challenge would be epic with this because I also use scaling health so I need your items to survive. Keep it up because this one is a must for me.

**Comment by sefldron** [comment]
Just to clarify, I meant that your mods are a must for me. :)

**Comment by SilentChaos512** [comment]
I do agree tools and armor are a bit too easy to maintain. Funny coincidence, I was just thinking of redesigning the tool system last night. I'll keep this in mind.

**Comment by sefldron** [comment]
Correct me if I'm wrong but are your tools treating the durability as a stack of the tool? Technically if a shovel had 200 durability that means it's virtually 200 one use shovels...

**Comment by SilentChaos512** [comment]
No, durability should behave as it does in vanilla.

**Comment by sefldron** [comment]
It does just fine but I think that is why the item name pops up every use of a tool...I see it so much but in the code it calls for a true or false if a block is cancelled during mining or similar values for weapons, when it "re-equips" it is actually replacing itself with itself. Seems like the same method of how you end up with just a stick when it breaks. It's still itself...every possible tool has a broken rod with it's name. The name outcome due to material configuration creates the id for reference to call it so it doesn't delete itself when it breaks...has anyone tested throwing one of your tools into lava ever to destroy it. I have coded with rpg maker on PC many times before so I understand how code constructs work but mc Java language is still new but the same logic would apply I believe.

**Comment by SilentChaos512** [comment]
The name pops up because the NBT of the tool is being modified with every block mined. Minecraft considers it a different stack each time because of that, even though it's not. The broken tools are also the exact same item, just with fewer layers rendered and the stats reduced to make it useless.
I'm not sure I follow you on the lava. Destroying the tool should delete it entirely. Everything that makes the tool what it is is in the NBT. Destroy the stack, and you destroy the NBT. Name, stats, statistics, everything.

**Comment by sefldron** [comment]
I was just curious about it from an uneducated standpoint. Knowledge is power and I'm slowly getting myself mentally ready to try to learn something new. It's like how I used to map for unreal tournament 2004 years ago and rpg maker. Those were easy though. Technically modding mc is like building a smaller game within the game that gets along with all the others. Thanks for that. I learned a lot from that to be honest.
